src/sysjmp.c

```diff
@@ -43,6 +43,32 @@ Int RegisterSyLongjmpObserver(voidfunc func)
     return 0;
 }
 
+enum { tryCatchFuncsLen = 16 };
+
+static TryCatchHandler tryCatchFuncs[tryCatchFuncsLen];
+
+Int RegisterTryCatchHandler(TryCatchHandler func)
+{
+    Int i;
+    for (i = 0; i < tryCatchFuncsLen; ++i) {
+        if (tryCatchFuncs[i] == func) {
+            return 1;
+        }
+        if (tryCatchFuncs[i] == 0) {
+            tryCatchFuncs[i] = func;
+            return 1;
+        }
+    }
+    return 0;
```

Comment:
The meaning of the return value of this function should be documented. Could also change it to use BOOL/TRUE/FALSE.

Reply:
The code is basically a carbon copy of `RegisterSyLongJmpObserver()`. I can easily incorporate all the changes; should I update `RegisterSyLongJmpObserver()` then too, while I'm at it?

Reply:
Just to say, I wrote RegisterSyLongJmpObserver, and I agree these are sensible changes to make there as well, if you are happy to do so.

Reply:
Okay, will do. :)